### PR TITLE
Floral Somatoray Gene Targeting

### PR DESCRIPTION
**Pull request by RustingWithYou**
Steals Baystation's specific gene firing mode for the floral somatoray. This should make it easier for xenobotany players to get to the interesting part of the job instead of playing at random guesswork and wasting all their disks on learning genes.

### CODE REVIEW

**Comment by RustingWithYou** [comment]
!review

**Review by NonQueueingMatt — APPROVED** [review]

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -40,6 +42,23 @@
...
+	set src in view(1)
+
+	var/genemask = input("Choose a gene to modify.") as null|anything in SSplants.plant_gene_datums
+
+	if(!genemask)
+		return
+
+	gene = SSplants.plant_gene_datums[genemask]
+
+	to_chat(usr, "<span class='info'>You set the [src]'s targeted genetic area to [genemask].</span>")
```
Comment (SleepyGemmy) [inline code/modules/projectiles/guns/energy/special.dm:57]:
```suggestion
	to_chat(usr, SPAN_INFO("You set \the [src]\s targeted genetic area to [genemask]."))
```

File: code/modules/projectiles/projectile/special.dm [hunk 2]
```diff
@@ -140,6 +140,14 @@
 	nodamage = 1
 	check_armor = "energy"
 
+/obj/item/projectile/energy/floramut/gene
+	name = "gamma somatoray"
+	icon_state = "energy2"
+	damage = 0
+	damage_type = TOX
+	nodamage = 1
```
Comment (SleepyGemmy) [inline code/modules/projectiles/projectile/special.dm:148]:
```suggestion
	nodamage = TRUE
```

File: code/modules/projectiles/guns/energy/special.dm [hunk 3]
```diff
@@ -40,6 +42,23 @@
...
+
+	return
+
+
```
Comment (SleepyGemmy) [inline code/modules/projectiles/guns/energy/special.dm:61]:
```suggestion
```
stray newline.

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**